go-runtime/goplugin/service.go

```diff
@@ -375,38 +229,41 @@ func build(ctx context.Context, projectConfig projectconfig.Config, stubsRoot st
 	return &langpb.BuildResponse{
 		Event: &langpb.BuildResponse_BuildSuccess{
 			BuildSuccess: &langpb.BuildSuccess{
-				ContextId:          buildCtx.ID,
-				IsAutomaticRebuild: isAutomaticRebuild,
-				Errors:             langpb.ErrorsToProto(buildErrs),
-				Module:             moduleProto,
-				Deploy:             deploy,
+				ModifiedFiles: captureTx.files,
```

Comment:
I dont think a list of files is enough here. We should capture the hashes at the time of modification by the build. And then feed those hashes back into the engine's watcher. Otherwise we miss out on file changes in this case:
- Build starts
- Build modifies a file
- User modifies the same file
- Build completes
A new build should start at this point

Reply:
I don't think it is worth worrying about this, as if a user is modifying files modified by the build while the build is running there are already races we can't guard against. In partular:
- If the user modifies the file slightly earlier the build just wipes out their changes
- Currently there is still a race between us modifying the contents and reading the hash, so there is still a window for a race to occur

In general modifying the files that the build modified while the build is in progress is going to give you a bad time.